File: src/main/java/net/hellomouse/progression_change/block/block_entity/BrickFurnaceBlockEntity.java
package net.hellomouse.progression_change.block.block_entity;

import net.hellomouse.progression_change.client.screen.BrickFurnaceScreenHandler;
import net.hellomouse.progression_change.registries.ProgressionModBlockEntityRegistry;
import net.minecraft.block.BlockState;
import net.minecraft.block.entity.AbstractFurnaceBlockEntity;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.entity.player.PlayerInventory;
import net.minecraft.recipe.RecipeType;
import net.minecraft.screen.ScreenHandler;
import net.minecraft.text.Text;
import net.minecraft.util.math.BlockPos;

public class BrickFurnaceBlockEntity extends AbstractFurnaceBlockEntity {
    public BrickFurnaceBlockEntity(BlockPos pos, BlockState state) {
        super(ProgressionModBlockEntityRegistry.BRICK_FURNACE.get(), pos, state, RecipeType.SMELTING);
    }

    @Override
    protected Text getContainerName() {
        return Text.translatable("container.progression_change.brick_furnace");
    }

    @Override
    protected ScreenHandler createScreenHandler(int syncId, PlayerInventory playerInventory) {
        return new BrickFurnaceScreenHandler(syncId, playerInventory, this, this.propertyDelegate);
    }

    @Override
    public void onOpen(PlayerEntity player) {
        super.onOpen(player);
        System.out.println("HELLO");
    }
}

File: src/main/kotlin/com/github/auri_f5bde6/xeno_early_start/block/block_entity/BrickFurnaceBlockEntity.kt
package com.github.auri_f5bde6.xeno_early_start.block.block_entity

import com.github.auri_f5bde6.xeno_early_start.client.screen.BrickFurnaceScreenHandler
import com.github.auri_f5bde6.xeno_early_start.registries.ProgressionModBlockEntityRegistry
import net.minecraft.block.BlockState
import net.minecraft.block.entity.AbstractFurnaceBlockEntity
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.recipe.RecipeType
import net.minecraft.screen.ScreenHandler
import net.minecraft.text.Text
import net.minecraft.util.math.BlockPos

class BrickFurnaceBlockEntity(pos: BlockPos, state: BlockState) :
    AbstractFurnaceBlockEntity(ProgressionModBlockEntityRegistry.BRICK_FURNACE.get(), pos, state, RecipeType.SMELTING) {
    override fun getContainerName(): Text {
        return Text.translatable("container.xeno_early_start.brick_furnace")
    }

    override fun createScreenHandler(syncId: Int, playerInventory: PlayerInventory): ScreenHandler {
        return BrickFurnaceScreenHandler(syncId, playerInventory, this, this.propertyDelegate)
    }

}
